lib/Doctrine/DBAL/Platforms/SQLServerPlatform.php

```diff
@@ -1532,6 +1534,20 @@ protected function initializeDoctrineTypeMappings()
         ];
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function isCommentedDoctrineType(Type $doctrineType)
+    {
+        if ($doctrineType->getName() === Types::TEXT) {
+            // We require a commented text type in order to distinguish between text and string
+            // as both (have to) map to the same native type.
+            return true;
```

Comment:
Only TEXT maps to `VARCHAR(MAX)`. Is it possible to identify this during the schema introspection? See https://github.com/doctrine/dbal/blob/5b456d8b77c3ea3fcb4b27b40d2cc60ea043cafd/src/Schema/SQLServerSchemaManager.php#L80-L86